AgiliTEAM/pacman_env.py

```diff
@@ -0,0 +1,282 @@
+            for j in range(self.image_size):
+                self.image[i][j] = self.state[i // self.ratio][j // self.ratio]
+
+    def _get_info(self) -> str:
+        """
+        This function creates an 'info' string which can be used as a feedback from the environment itself.
+        :return: information about the game's current state
+        """
+        self._calculate_score()
+        info = "Points acquired: " + str(self.score) + "\nSteps: " + str(self.step_counter) + "/" + str(MAX_STEP)
+
+        return info
+
+    def _get_reward(self, state: np.ndarray) -> float:
+        """
+        This function realizes the reward function and formulates the value based on the state passed as an argument.
+        :param state: ndarray of the game state based on which the reward value is calculated
+        :return: reward value from which the agent is able to learn
+        """
+        # Only for RL purposes
+        # To be implemented later...
+        reward = 0.0
+
+        return reward
+
+    def _get_next_state(self, action: int) -> np.ndarray:
+        """
+        This function realizes the change in the game state according to the chosen action.
+        :param action: chosen action 0/1/2/3 corresponding to the directions in order up/right/down/left)
+        :return: state of the environment after the action
+        """
+        self._check_action_validity(action)
+        self._set_action(action)
+        self._move()
+        next_state = self._update_map()
+
+        return next_state
+
+    def _move(self) -> None:
+        """
+        This function modifies PacMan's position according to its facing direction.
+        :return: None
+        """
+        # if facing of the pacman == _direction_:
+        #     if pacman is at the end of the map:
+        #         move to the opposite side
+        #     else:
+        #         move a unit in the _direction_
+        if self.orientation == DIRECTIONS["up"]:
+            self.pos[0] = MAP_SIZE - 1 if self.pos[0] - 1 < 0 else self.pos[0] - 1
+        elif self.orientation == DIRECTIONS["right"]:
+            self.pos[1] = 0 if self.pos[1] + 1 == MAP_SIZE else self.pos[1] + 1
+        elif self.orientation == DIRECTIONS["down"]:
+            self.pos[0] = 0 if self.pos[0] + 1 == MAP_SIZE else self.pos[0] + 1
+        elif self.orientation == DIRECTIONS["left"]:
+            self.pos[1] = MAP_SIZE - 1 if self.pos[1] - 1 < 0 else self.pos[1] - 1
+
+        self.step_counter += 1
+
+    def _set_action(self, action: int | None) -> None:
+        """
+        This function coordinates the modification of the 'orientation' variable based on user and timer (To be
+        implemented...) input.
+        :param action: chosen action 0/1/2/3 corresponding to the directions in order up/right/down/left)
+        :return: None
+        """
+        self.orientation = action
+        # Timer response to be implemented later...
+
+    def _check_action_validity(self, action: int) -> None:
```

Comment:
Check_Action_validity function could return a boolean value instead of None due to future improvement plans.